module/solutions/helper.js

```diff
@@ -84,4 +91,362 @@ module.exports = class SolutionsHelper {
     });
   }
 
+     /**
+   * Create solution.
+   * @method 
+   * @name createSolution
+   * @param {Object} data - solution creation data.
+   * @returns {JSON} solution creation data. 
+   */
+  
+  static createSolution(data) {
+    return new Promise(async (resolve, reject) => {
+        try {
+
+            if( data.programExternalId ) {
+
+              let programData = await programsHelper.programDocuments({
+                externalId : data.programExternalId
+              },["name","description"]);
+
+              if ( !programData.length > 0 ) {
+                throw {
+                  message : constants.apiResponses.PROGRAM_NOT_FOUND
+                }
+              }
+
+              data.programId = programData[0]._id;
+              data.programName = programData[0].name;
+              data.programDescription = programData[0].description; 
+            }
+
+            if( data.entityType ) {
+              
+              let entityTypeData = 
+              await entityTypesHelper.entityTypesDocument({
+                name : data.entityType
+              },["_id"]);
+
+              if( !entityTypeData.length > 0 ) {
+                throw {
+                  message : constants.apiResponses.ENTITY_TYPES_NOT_FOUND
+                }
+              }
+            }
+
+            if( data.entities && data.entities.length > 0 ) {
```

Comment:
@aman-tunerlabs - This will have to be removed based on if entities are not required to be set in imp project solution